src/app/progress/progress/progress.component.ts

```diff
@@ -171,7 +178,21 @@ export class ProgressComponent implements OnInit {
     }
   }
 
-  averageStudentMark(servData) {
+  getAverageStudentMark(servData) {
+    let avrStudentMark = 0;
+
+    // find average student mark by subject
+    if (servData.length !== 0) {
```

Comment:
no need to check length, arrays methods will not be run if length is 0

Reply:
corrected it